### ISSUE TITLE
Inconsistent behavior of setMaxWorkerExecuteTime in DeploymentOptions depending on worker pool name

### ISSUE THREAD

**Issue by VikramVuppla** [eclipse-vertx/vert.x#4690]
### Questions

Do not use this issue tracker to ask questions, instead use one of these [channels](https://vertx.io/community/). Questions will likely be closed without notice.

### Version

4.3.8

### Context

Inconsistent behavior when trying to `setMaxWorkerExecuteTime` using `DeploymentOptions`, if `workerPoolName` is set then `setMaxWorkerExecuteTime` is honored but if it is not set then BlockedThreadChecker is throwing `Thread[vert.x-worker-thread-5,5,main] has been blocked for 60999 ms, time limit is 60000 ms` warning.

Surprisingly if `setMaxWorkerExecuteTime` is set using `VertxOptions` it is honored as well i.e., `vertxOptions.setMaxWorkerExecuteTime`

### Do you have a reproducer?

A reproducer is a simple project hosted on GitHub (or another forge supporting git clone operation) that has a build file that can be executed to reproduce the issue.

Reproducers are very helpful for contributors and will likely help them fixing your bug faster.

* Link to github project/gist

#### Failure case
1. Define DeploymentOptions and setWorker to true, setMaxWorkerExecuteTime to 2 mins
```
DeploymentOptions deploymentOptions = new DeploymentOptions()
                .setWorker(true)
                .setMaxWorkerExecuteTime(2)
                .setMaxWorkerExecuteTimeUnit(TimeUnit.MINUTES)
```
2. Deploy the verticle using above options
```
Vertx.vertx().deployVerticle(new MyWorkerVerticle(), deploymentOptions)
```

if a blocking code is executed which takes more than 60s in `MyWorkerVerticle` then it throws a warning i.e.,

`Thread[vert.x-worker-thread-5,5,main] has been blocked for 60999 ms, time limit is 60000 ms`

#### Success case 1
1. Define DeploymentOptions and setWorker to true, setMaxWorkerExecuteTime to 2 mins, give a workerPoolName using setWorkerPoolName
```
DeploymentOptions deploymentOptions = new DeploymentOptions()
                .setWorker(true)
                .setWorkerPoolName("my-worker-pool")
                .setMaxWorkerExecuteTime(2)
                .setMaxWorkerExecuteTimeUnit(TimeUnit.MINUTES)
```
2. Deploy the verticle using above options
```
Vertx.vertx().deployVerticle(new MyWorkerVerticle(), deploymentOptions)
```
then `setMaxWorkerExecuteTime` is honored and `thread blocked` warning is not logged if a blocking code is executed above 60s and below 120s.

#### Success case 2
1. Define VertxOptions and setMaxWorkerExecuteTime to 2 mins
```
        VertxOptions vertxOptions = new VertxOptions();
        vertxOptions.setMaxWorkerExecuteTime(10).setMaxWorkerExecuteTimeUnit(TimeUnit.MINUTES);
```
2. Deploy the verticle using above options
```
Vertx.vertx(vertxOptions).deployVerticle(new MyWorkerVerticle())
```
then `setMaxWorkerExecuteTime` is honored and `thread blocked` warning is not logged if a blocking code is executed above 60s and below 120s.

### Extra

* Anything that can be relevant such as OS version, JVM version

**Comment by tsegismont** [comment]
The behavior is correct but should be documented in Javadoc, like it has been regarding pool size:

https://github.com/eclipse-vertx/vert.x/blob/f9956fbd512d8c158a0522ad7d62ed8b3fb13edb/src/main/java/io/vertx/core/DeploymentOptions.java#L292-L293

**Comment by tsegismont** [comment]
Closed in b243878c388ad67a27c3a36a5c0c525afeeec947
